### ISSUE TITLE
[Accessibility Issues from Teams Mobile] - Random Order in Reading out Elements

### ISSUE THREAD

**Issue by VikrantSingh01** [microsoft/AdaptiveCards#3712]
# Platform

What platform is your issue or question related to? (Delete other platforms).
* iOS

# Author or host
Teams Mobile Client.

# Version of SDK
1.2.5

# Details

Issue : 
Creating Another bug to track the Pending Accessibility Issues from https://github.com/microsoft/AdaptiveCards/issues/3271

The main issue is the **order of the reading out the accessibility elements is random** on iOS.

Internal Teams Work Item : 
https://domoreexp.visualstudio.com/MSTeams/_workitems/edit/607780

CC - @nesalang @shalinijoshi19  @arpitgupta19 @Aditya0603  @siduppal

**Comment by VikrantSingh01** [comment]
Action Plan :
@nesalang to add loggers to the Adaptive Card SDK and share the logging enabled Binary to debug the accessibility from our Client. 
@Aditya0603  / @shbhadra : To Share the Teams (with instrumented Adaptive Card Library) .apk and .ipa and .app to run the Android and iOS Simulator. (with Instruction)

**Comment by shalinijoshi19** [comment]
@nesalang how are we looking on the private framework package we needed to provide to the teams team for debugging?

**Comment by nesalang** [comment]
> @nesalang how are we looking on the private framework package we needed to provide to the teams team for debugging?

i'm checking out apple's apis for accessibility information. hoping to create a PR today.

**Comment by shalinijoshi19** [comment]
@nesalang has a PR out https://github.com/microsoft/AdaptiveCards/pull/3727 that should allow additional debugging of AC elements. @VikrantSingh01 to follow up with @Aditya0603 on current status.

**Comment by VikrantSingh01** [comment]
@nesalang / @shalinijoshi19 : Kindly let us know once this accessibility fix is out. We would like to do integrate this along with frictionless update P.R (via Cocopad). CC - @Aditya0603 / @shbhadra

**Comment by shalinijoshi19** [comment]
@VikrantSingh01 release/1.2 branch has the changes merged in a this point. @nesalang is there a framework file that we can share with Teams built out of hte code currently in release/1.2? Thanks!

**Comment by nesalang** [comment]
> @VikrantSingh01 release/1.2 branch has the changes merged in a this point. @nesalang is there a framework file that we can share with Teams built out of hte code currently in release/1.2? Thanks!

@shalinijoshi19 & @VikrantSingh01 [here is the link](https://microsoft.visualstudio.com/OS/_build/results?buildId=21034196&view=artifacts) to the artifact.

**Comment by VikrantSingh01** [comment]
@arpitgupta19 : Can we get this verified with the list of accessibility issues we have in our Mobile queue

**Comment by nesalang** [comment]
@nesalang check on the status with the accessibility log addition.

**Comment by shalinijoshi19** [comment]
@vikrantsingh01 could you help follow up on this one to see if the additional logging support here helped ?

**Comment by jwoo-msft** [comment]
@jwoo-msft --> update this issue

**Comment by adumrewal** [comment]
@jwoo-msft, @shalinijoshi19 : I can still experience this issue with SDK 1.2.7 on Teams iOS. For example, if a card has ActionSet/ChoiceSet (compact style), the order is not being followed correctly.
[VoiceOverVideo](https://microsoft-my.sharepoint.com/:v:/p/amdumrew/EXgEzOq4DVNCsCfdrQuNAIIBkFxBzyVAnItYAvoWdObDCw?e=BQLcab)

**Comment by jwoo-msft** [comment]
@adumrewal, could you please repro this issue with same payload and accessibility inspector enabled, and share the log you collect?
@anbah34

Thanks

**Comment by shalinijoshi19** [comment]
@VikrantSingh01 / @anbah34 / @adumrewal ping. Thanks

**Comment by jwoo-msft** [comment]
@adumrewal 
hi,

Can you share the payload shown in your video, so I can try it on my end? With the payload I have, I couldn't reproduce the issue with main branch.

Can you also share a video with audio? It's really hard to figure out what's the issue. Please also provide steps you've taken to reproduce the issue. 

I think we should only concerned about one card at a time. In the video, the focus was moving to another card, that's not something AdaptiveCard SDK can control. You might have to set the focus again on the row that comes into the view.

Thanks

**Comment by shalinijoshi19** [comment]
Closing this out for now as it isn't a 1.3 upgrade blocker and we havent had traction on this for a while. @VikrantSingh01 / @adumrewal can you reactivate if this still continues to repro with latest bits pls? 
Thanks

**Comment by adumrewal** [comment]
I tested this with the latest main branch and I'm unable to repro it there. Looks like a Teams client specific issue.

**Comment by jwoo-msft** [comment]
> I tested this with the latest main branch and I'm unable to repro it there. Looks like a Teams client specific issue.

cc @shalinijoshi19
